File: algorithms-data-structures/ts/data-structures/stacks/min-stack/min-stack.test.ts
import {MinStack} from "./min-stack";
import {expect, test} from "bun:test";

test("Minstack", () => {
  let minStack = new MinStack();
  minStack.push(1);
  minStack.push(-2);
  minStack.push(3);
  expect(minStack.getMin()).toBe(-2);
  minStack.pop();
  expect(minStack.top()).toBe(-2);
  expect(minStack.getMin()).toBe(-2);
  minStack.pop();
  expect(minStack.getMin()).toBe(1);
});

File: algorithms-data-structures/go/src/data-structures/stacks/min-stack/min-stack_test.go
package minstack

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestMinStack(t *testing.T) {
	m := Constructor()
	m.Push(-2)
	m.Push(0)
	m.Push(-3)
	min := m.GetMin()
	assert.Equal(t, -3, min)
	m.Pop()
	top := m.Top()
	assert.Equal(t, 0, top)
	min = m.GetMin()
	assert.Equal(t, -2, min)
}
